packages/core/src/controllers/SendController.ts

```diff
@@ -93,49 +94,85 @@ export const SendController = {
   },
 
   sendToken() {
-    if (this.state.token?.address && this.state.sendTokenAmount && this.state.receiverAddress) {
-      EventsController.sendEvent({
-        type: 'track',
-        event: 'SEND_INITIATED',
-        properties: {
-          isSmartAccount:
-            AccountController.state.preferredAccountType ===
-            W3mFrameRpcConstants.ACCOUNT_TYPES.SMART_ACCOUNT,
-          token: this.state.token.address,
-          amount: this.state.sendTokenAmount,
-          network: NetworkController.state.caipNetwork?.id || ''
+    switch (ChainController.state.activeCaipNetwork?.chain) {
+      case 'evm':
+        if (this.state.token?.address && this.state.sendTokenAmount && this.state.receiverAddress) {
+          EventsController.sendEvent({
+            type: 'track',
+            event: 'SEND_INITIATED',
+            properties: {
+              isSmartAccount:
+                AccountController.state.preferredAccountType ===
+                W3mFrameRpcConstants.ACCOUNT_TYPES.SMART_ACCOUNT,
+              token: this.state.token.address,
+              amount: this.state.sendTokenAmount,
+              network: NetworkController.state.caipNetwork?.id || ''
+            }
+          })
+          this.sendERC20Token({
+            receiverAddress: this.state.receiverAddress,
+            tokenAddress: this.state.token.address,
+            sendTokenAmount: this.state.sendTokenAmount,
+            decimals: this.state.token.quantity.decimals
+          })
+        } else if (
+          this.state.receiverAddress &&
+          this.state.sendTokenAmount &&
+          this.state.gasPrice &&
+          this.state.token?.quantity.decimals
+        ) {
+          EventsController.sendEvent({
+            type: 'track',
+            event: 'SEND_INITIATED',
```

Comment:
Abstract this to `SendController.sendEvmToken()` and `SendController.sendSolanaToken` or similar

Reply:
https://github.com/WalletConnect/web3modal/pull/2802/commits/53b5c63dc47de147039789d4d2458e891ef576b0